File: raincoat-fabric/src/main/java/dev/uten2c/raincoat/States.java
package dev.uten2c.raincoat;

import org.jetbrains.annotations.Nullable;

public final class States {
    private static boolean isOnServer = false;
    public static @Nullable Long directionSendRequestedTime = null;

    private States() {
    }

    public static void reset() {
        isOnServer = false;
        directionSendRequestedTime = null;
    }

    public static boolean isOnServer() {
        return isOnServer;
    }

    public static void onJoinServer() {
        isOnServer = true;
    }
}

File: raincoat-fabric/src/main/kotlin/dev/uten2c/raincoat/States.kt
package dev.uten2c.raincoat

import dev.uten2c.raincoat.debug.DebugShapes
import dev.uten2c.raincoat.shake.ShakeEffect

object States {
    @JvmStatic
    var isOnServer = false
        private set

    @JvmStatic
    var isHandshakeReceived = false

    var directionSendRequestedTime: Long? = null

    @JvmStatic
    var isRecoiling = false

    @JvmStatic
    var showDebugShape = false

    @JvmStatic
    var shakeEffect: ShakeEffect? = null

    fun reset() {
        isOnServer = false
        isHandshakeReceived = false
        directionSendRequestedTime = null
        isRecoiling = false
        shakeEffect = null
        DebugShapes.clearShapes()
    }

    fun onJoinServer() {
        isOnServer = true
    }
}
